fili-core/src/test/groovy/com/yahoo/bard/webservice/web/handlers/CacheWeightCheckRequestHandlerSpec.groovy

```diff
@@ -0,0 +1,260 @@
+        response = Mock(WeightCheckResponseProcessor)
+        bardQueryInfo = BardQueryInfoUtils.initializeBardQueryInfo()
+        json = new JsonNodeFactory().arrayNode()
+        cacheService = new QuerySignedCacheService(dataCache, querySigningService, MAPPER)
+        handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        )
+    }
+
+    def cleanup() {
+        BardQueryInfoUtils.resetBardQueryInfo()
+    }
+
+    def "Check constructor"() {
+        expect:
+        handler.next == next
+        handler.webService == webService
+        handler.queryWeightUtil == queryWeightUtil
+        handler.mapper == MAPPER
+        handler.querySignedCacheService == cacheService
+    }
+
+    def "Request passing quick weight check"() {
+        setup:
+        1 * queryWeightUtil.skipWeightCheckQuery(groupByQuery) >> true
+        1 * next.handleRequest(context, request, groupByQuery, response) >> true
+
+        expect:
+        handler.handleRequest(context, request, groupByQuery, response)
+
+        and:
+        bardQueryInfo.queryCounter.get(BardQueryInfo.WEIGHT_CHECK).get() == 0
+    }
+
+    def "Request handled without building callback"() {
+        setup:
+        final SuccessCallback success = Mock(SuccessCallback)
+        groupByQuery.getGranularity() >> DAY
+        groupByQuery.getInnermostQuery() >> groupByQuery
+        WeightEvaluationQuery weightQuery = Mock(WeightEvaluationQuery)
+        queryWeightUtil.makeWeightEvaluationQuery(groupByQuery) >> weightQuery
+        writer.writeValueAsString(weightQuery) >> "Weight query"
+        CacheWeightCheckRequestHandler handler = new CacheWeightCheckRequestHandler(
+                next,
+                webService,
+                queryWeightUtil,
+                MAPPER,
+                cacheService
+        ) {
+            @Override
+            public SuccessCallback buildCacheSuccessCallback(
+                    final SuccessCallback classicSuccessCallback,
+                    final QuerySignedCacheService cacheService,
+                    final DruidAggregationQuery<?> groupByQuery,
+                    final ResponseProcessor response
+            ) {
+                groupByQuery.clone()
+                return success
+            }
+        }
+
+        1 * queryWeightUtil.skipWeightCheckQuery(groupByQuery) >> false
+        0 * next.handleRequest(context, request, groupByQuery, response) >> true
+        1 * queryWeightUtil.getQueryWeightThreshold(DAY) >> 5
+        1 * groupByQuery.clone() >> groupByQuery
+        1 * webService.postDruidQuery(context, success, null, null, weightQuery)
+        1 * response.getErrorCallback(groupByQuery)
+        1 * response.getFailureCallback(groupByQuery)
+
+        expect:
+        handler.handleRequest(context, request, groupByQuery, response)
```

Comment:
This looks like you're just running your own handler that you created in this test.  Am I missing something?